x-pack/agent/pkg/agent/application/emitter.go

```diff
@@ -33,6 +39,12 @@ func emitter(log *logger.Logger, router *router, decorators ...decoratorFunc) em
 			return errors.New(err, "could not create the AST from the configuration", errors.TypeConfig)
 		}
 
+		for _, filter := range modifiers.Filters {
+			if err := filter(ast); err != nil {
+				return errors.New(err, "failed to filter configuration", errors.TypeConfig)
+			}
+		}
```

Comment:
Maybe we should log when a constrains fails?

Reply:
I think I did not correctly express my requirement here, my bad. 

The above code do log when a constraint fails, but I think we should also log when a constraints matches and we filter content.